client/extensions/woocommerce/state/sites/settings/stripe-connect-account/actions.js

```diff
@@ -220,3 +224,124 @@ function deauthorizeFailure( siteId, action, errorMessage ) {
 		siteId,
 	};
 }
+
+/**
+ * Action Creator: Get the initial OAuth URL for connecting a Stripe Account.
+ *
+ * @param {Number} siteId The id of the site for which to create an account.
+ * @param {String} returnUrl The URL for Stripe to return the user to (to complete the setup)
+ * @param {String} [successAction=undefined] Optional action object to be dispatched upon success.
+ * @param {String} [failureAction=undefined] Optional action object to be dispatched upon error.
+ * @return {Object} Action object
+ */
+export const oauthInit = ( siteId, returnUrl, successAction = null, failureAction = null ) => (
+	dispatch,
+	getState
+) => {
+	const state = getState();
+	if ( ! siteId ) {
+		siteId = getSelectedSiteId( state );
+	}
+
+	const initAction = {
+		type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_OAUTH_INIT,
+		returnUrl,
+		siteId,
+	};
+
+	dispatch( initAction );
+
```

Comment:
Do we need to be concerned with validating `returnUrl` has been supplied and formatted properly?

Reply:
We could validate that it looks like a URL, but that's about it unfortunately. How about in a follow on issue - we can ruggedize all the action creators with argument validation.

Reply:
https://github.com/Automattic/wp-calypso/issues/19352